### PR TITLE
Use $strict param in functions that have it

### PR DESCRIPTION
**Pull request by carusogabriel**
Some functions, e.g. `in_array`, have a third param to enable strict comparison :shield:

### CODE REVIEW

File: lib/Doctrine/DBAL/DriverManager.php [hunk 1]
```diff
@@ -204,7 +204,7 @@ private static function _checkParams(array $params): void
             throw DBALException::unknownDriver($params['driver'], array_keys(self::$_driverMap));
         }
 
-        if (isset($params['driverClass']) && ! in_array('Doctrine\DBAL\Driver', class_implements($params['driverClass'], true))) {
+        if (isset($params['driverClass']) && ! in_array('Doctrine\DBAL\Driver', class_implements($params['driverClass'], true), true)) {
```
Comment (morozov) [inline lib/Doctrine/DBAL/DriverManager.php]: I'm not a fan of complicating code for no reason. In this case, both elements of the `$haystack` and `$needle` are strings. What's the point of changing this code?

I agree strict comparison can be used when the types are not clear/enforced, but it's not applicable to many cases in  this pull request.

Comment (carusogabriel) [reply lib/Doctrine/DBAL/DriverManager.php]: @Ocramius So should we wait to add [phpstan/phpstan-strict-rules](https://github.com/phpstan/phpstan-strict-rules)?

Comment (Ocramius) [reply lib/Doctrine/DBAL/DriverManager.php]: Possibly, yes. That would take a lot of work to integrate though.

**Comment by carusogabriel** [comment]
Closing as we will wait to integrate with [phpstan/phpstan-strict-rules](https://github.com/phpstan/phpstan-strict-rules)

**Pull request closed without merging**